MoltenVK/MoltenVK/GPUObjects/MVKImage.mm

```diff
@@ -1552,14 +1552,14 @@ static void signalAndUntrack(const MVKSwapchainSignaler& signaler) {
 	id<CAMetalDrawable> mtlDrwbl = getCAMetalDrawable();
 	MVKSwapchainSignaler signaler = getPresentationSignaler();
 	[mtlCmdBuff addScheduledHandler: ^(id<MTLCommandBuffer> mcb) {
-
-		addPresentedHandler(mtlDrwbl, presentInfo, signaler);
-
 		// Try to do any present mode transitions as late as possible in an attempt
 		// to avoid visual disruptions on any presents already on the queue.
 		if (presentInfo.presentMode != VK_PRESENT_MODE_MAX_ENUM_KHR) {
 			mtlDrwbl.layer.displaySyncEnabledMVK = (presentInfo.presentMode != VK_PRESENT_MODE_IMMEDIATE_KHR);
 		}
+		if (getEnabledExtensions().vk_GOOGLE_display_timing.enabled) {
```

Comment:
In doing another review, I've just realized something that I didn't catch in our earlier discussions here.

`addPresentedHandler(), along with beginPresentation() and endPresentation()` do more than just track `VK_GOOGLE_display_timing`. They also:

- `retain()/release()` the swapchain image, which is important for ensuring the image doesn't get destroyed before the presentation finishes.
- Signal the presentation fence
- Signal presentation semaphores that don't use `MTLEvent`
- Track presentation performance statistics for MoltenVK

See the f0cb31a  commit comments for more discussion.

Removing `hasPresentTime`, and the new calculations for `presentMargin`, are both useful here, but I think the original idea of reintroducing a guard on `addPresentedHandler` may regress several previous sync and race-conditions issue fixes.

Reply:
Thanks for remembering this.  Darn - I should have at least noticed the performance statistics being part of the presentHandler logic.  Glad you caught this.  I guess we will have to live with folks potentially running into the uncaught exception when Xcode GPU frame capture is default on with debug builds.  Unless of course you can find a solution with Apple or some other workaround.

I have reverted the guard, and renamed `startPresentTime` to `beginPresentTime`.  I think this more closely matches the naming nomenclature of the begin/end presentation routines, and differentiates it more clearly from the performance statistics variable `_presentationStartTime`.

Reply:
> I guess we will have to live with folks potentially running into the uncaught exception when Xcode GPU frame capture is default on with debug builds.

TBH...I run GPU capture all the time during dev & testing, and I have never encountered an issue. Nor did the errors make any kind of sense. To me, that original problem feels like a complex environmental issue of some kind (OS, Xcode, app usage of MoltenVK, etc). Until we get some further replaceable data on it, bypassing legit functionality to avoid it seems like dangerous overkill. Let's see what happens from further users.

Reply:
I have been able to reproduce it.  Let me see if I can document the conditions to help you with diagnosis.  I will put my comments in the original reported issue.  Hopefully this PR is now refocussed and can proceed.